### PR TITLE
add lldb-specific HostIoOpenFlags

### PR DESCRIPTION
**Pull request by mrk-its**
### Description

This PR adds lldb-specific HostIoOpenFlags, as proposed in #99

### API Stability

- [x] This PR does not require a breaking API change

### Checklist

- Implementation
  - [x] `cargo build` compiles without `errors` or `warnings`
  - [x] `cargo clippy` runs without `errors` or `warnings`
  - [x] `cargo fmt` was run
  - [x] All tests pass
- Documentation
  - [x] rustdoc + approprate inline code comments
  - [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by mrk-its** [comment]
second try :)

File: src/target/ext/host_io.rs [hunk 1]
```diff
@@ -24,6 +24,18 @@ bitflags! {
         const O_TRUNC = 0x400;
         /// Exclusive access.
         const O_EXCL = 0x800;
+
+        // LLDB-specific flags, as documented at:
+        // https://github.com/llvm/llvm-project/blob/ec642ceebc1aacc8b16249df7734b8cf90ae2963/lldb/include/lldb/Host/File.h#L47-L66
```
Comment (daniel5151) [inline src/target/ext/host_io.rs:29]: Please lift this comment into the doc comment.

File: src/target/ext/host_io.rs [hunk 2]
```diff
@@ -24,6 +24,18 @@ bitflags! {
...
+
+        /// Do not block
```
Comment (daniel5151) [inline src/target/ext/host_io.rs:31]: Can you prefix these doc comments with `(LLDB extension)`?

Oh, and end each comment with a `.` (for consistency)

File: src/target/ext/host_io.rs [hunk 3]
```diff
@@ -25,16 +25,16 @@ bitflags! {
         /// Exclusive access.
         const O_EXCL = 0x800;
 
-        // LLDB-specific flags, as documented at:
-        // https://github.com/llvm/llvm-project/blob/ec642ceebc1aacc8b16249df7734b8cf90ae2963/lldb/include/lldb/Host/File.h#L47-L66
+        /// LLDB-specific flags, as documented at:
+        /// https://github.com/llvm/llvm-project/blob/ec642ceebc1aacc8b16249df7734b8cf90ae2963/lldb/include/lldb/Host/File.h#L47-L66
```
Comment (daniel5151) [inline src/target/ext/host_io.rs:29]: Sorry, I think you misunderstood - this doc comment needs to be lifted up to the main `HostIoOpenFlags` doc comment, after line `11`.

Comment (mrk-its) [reply src/target/ext/host_io.rs]: makes sense :)

File: src/target/ext/host_io.rs [hunk 4]
```diff
@@ -9,6 +9,7 @@ bitflags! {
     ///
     /// Extracted from the GDB documentation at
     /// [Open Flags](https://sourceware.org/gdb/current/onlinedocs/gdb/Open-Flags.html#Open-Flags)
+    /// [LLDB-specific Flags](https://github.com/llvm/llvm-project/blob/ec642ceebc1aacc8b16249df7734b8cf90ae2963/lldb/include/lldb/Host/File.h#L47-L66)
```
Comment (daniel5151) [inline src/target/ext/host_io.rs:12]: Sorry to keep throwing nits your way, but if you run `cargo doc`, you'll find that the documentation will look quite weird the way its currently written:

> Extracted from the GDB documentation at *Open Flags* *LLDB-specific Flags*

I'd probably do something like:

```rust
    /// Extracted from the GDB documentation at
    /// [Open Flags](https://sourceware.org/gdb/current/onlinedocs/gdb/Open-Flags.html#Open-Flags),
    /// and the LLDB source code at 
    /// [`lldb/include/lldb/Host/File.h`](https://github.com/llvm/llvm-project/blob/ec642ceebc1aacc8b16249df7734b8cf90ae2963/lldb/include/lldb/Host/File.h#L47-L66)
```

Comment (mrk-its) [reply src/target/ext/host_io.rs:12]: Np, I should have checked it out. Fixing

Comment (mrk-its) [reply src/target/ext/host_io.rs]: Done. It looks much better now :)

**Review by daniel5151 — APPROVED** [review]
Awesome - thanks for the PR 😄 

I'll push 0.6.1 up to crates.io after work today

**Comment by daniel5151** [comment]
0.6.1 is published to crates.io

**Pull request merged**